autoload/vimanzo/query.vim

```diff
@@ -26,23 +26,40 @@ function! vimanzo#query#GetGraph(uri)
     execute "!" . g:anzo_command . " get -z " . g:anzo_settings . " " . a:uri 
 endfunction
 
-
-"This function runs a query against the journal showing the results in a minibuffer
-function! vimanzo#query#ExecuteJournalQuery()
+"This is a generic function that runs a query and puts the results into
+"minibuffer
+function! vimanzo#query#ExecuteQuery(query_options)
   let l:query_file = bufname("%")
   silent! exe "noautocmd botright pedit QueryResults"
   noautocmd wincmd P
   set buftype=nofile
+  set nowrap
   exec ":norm ggdG"
-  exec ": read ! " . g:anzo_command . " query -a -z " . g:anzo_settings . " -f " . l:query_file
+  exec ": read ! " . g:anzo_command . " query " . a:query_options . " -z " . g:anzo_settings . " -f " . l:query_file
   noautocmd wincmd p
 endfunction
 
-com! JournalQuery :call vimanzo#query#ExecuteJournalQuery()
+"This function runs a query against the journal showing the results in a minibuffer
+function! vimanzo#query#ExecuteJournalQuery()
+  call vimanzo#query#ExecuteQuery("-a")
+endfunction
 
-function! vimanzo#query#ExecuteQuery()
-    silent !clear
-    execute "!" . g:anzo_command . " query -z " . g:anzo_settings . " -f " . bufname("%")
+"This function executes a sparql query and unpacks the json
+"to reutrn the results as a vim structure
+function! vimanzo#query#queryForVim()
+  let l:query_file = bufname("%")
+  " Use json results and all graphs
+  let l:query_options="-a -o json"
```

Comment:
One thing that might stop things up here is that I think this option errors if you have a construct query.

Reply:
true -- this is only for select and probably just for internal functions.  After I merge your other PR I'll make an issue to refactor the graphmart query to use this.